File: jobs/Mobile/packages/movie_repository/lib/src/models/movie.dart
import 'package:equatable/equatable.dart';
import 'package:json_annotation/json_annotation.dart';

part 'movie.g.dart';

@JsonSerializable()
class MovieDetails extends Equatable {
  const MovieDetails({
    required this.adult,
    required this.budget,
    required this.genres,
    required this.id,
    this.posterPath,
    this.backdrop,
    required this.title,
    this.overview,
    required this.releaseDate,
    this.runtime,
    required this.voteAverage,
    required this.voteCount,
  });

  final bool adult;
  final int budget;
  final List<String> genres;
  final int id;
  final String? posterPath;
  final String? backdrop;
  final String title;
  final String? overview;
  final DateTime? releaseDate;
  final int? runtime;
  final double voteAverage;
  final int voteCount;

  factory MovieDetails.fromJson(json) => _$MovieFromJson(json);

  Map<String, dynamic> toJson() => _$MovieToJson(this);

  @override
  List<Object?> get props => [
        adult,
        budget,
        genres,
        id,
        posterPath,
        backdrop,
        title,
        overview,
        releaseDate,
        runtime,
        voteAverage,
        voteCount
      ];
}

File: src/types/movie.ts
export interface Movie {
  id: number
  title: string
  overview: string
  poster_path: string | null
  backdrop_path: string | null
  release_date: string
  vote_average: number
  vote_count: number
  genre_ids?: number[]
  adult?: boolean
  original_language?: string
  original_title?: string
  popularity?: number
  video?: boolean
}

export interface MovieDetails extends Movie {
  genres?: Genre[]
  runtime?: number
  production_companies?: ProductionCompany[]
  belongs_to_collection?: Collection | null
  budget?: number
  homepage?: string
  imdb_id?: string
  origin_country?: string[]
  production_countries?: Country[]
  revenue?: number
  spoken_languages?: SpokenLanguage[]
  status?: string
  tagline?: string
}

export interface Genre {
  id: number
  name: string
}

export interface ProductionCompany {
  id: number
  name: string
  logo_path?: string | null
  origin_country?: string
}

export interface Collection {
  id: number
  name: string
  poster_path: string | null
  backdrop_path: string | null
}

export interface Country {
  iso_3166_1: string
  name: string
}

export interface SpokenLanguage {
  english_name: string
  iso_639_1: string
  name: string
}

export interface PaginatedResponse<T> {
  page: number
  results: T[]
  total_pages: number
  total_results: number
}

export type MovieSearchResponse = PaginatedResponse<Movie>
export type MovieListResponse = PaginatedResponse<Movie>
